internal/auth/ldap/repository_auth_method_read.go

```diff
@@ -32,19 +34,36 @@ func (r *Repository) LookupAuthMethod(ctx context.Context, publicId string, opt
 	return r.lookupAuthMethod(ctx, publicId, WithUnauthenticatedUser(ctx, opts.withUnauthenticatedUser))
 }
 
-// ListAuthMethods returns a slice of AuthMethods for the scopeId. The
-// WithUnauthenticatedUser, WithLimit and WithOrder options are supported and
-// all other options are ignored.
-func (r *Repository) ListAuthMethods(ctx context.Context, scopeIds []string, opt ...Option) ([]*AuthMethod, error) {
+// ListAuthMethods returns a slice of AuthMethods for the scopeId. Supported options:
+//   - auth.WithLimit
+//   - auth.WithStartPageAfterItem
+func (r *Repository) ListAuthMethods(ctx context.Context, scopeIds []string, opt ...auth.Option) ([]auth.AuthMethod, error) {
```

Comment:
The only way to allow us to call into the repositories from the `auth` package is to avoid any types defined in the subpackages (or we'd have import cycles). That is why we're using `auth.Option` and `auth.AuthMethod` here.